.gitignore

```diff
@@ -1,17 +1,20 @@
-*~
-*.pyc
-*.pyo
-*.swp
-parser.out
-parsetab.py
-.idea
+/.cache
 /.tox
+/.tox-*
+/.venv
+/.venv-*
```

Comment:
I added the above three entries. The other changes to this file are just a reordering of existing entries.